src/nvim/normal.c

```diff
@@ -846,7 +846,7 @@ normal_cmd (
        * mapping. */
       no_mapping--;
       while (enc_utf8 && lang && (c = vpeekc()) > 0
-             && (c >= 0x100 || MB_BYTE2LEN(vpeekc()) > 1)) {
+             && (c >= 0x100 || mb_byte2len(c) > 1)) {
```

Comment:
If this is the only -Wconversion error added then I think you need to have a cast here (`(uint8_t) c`). Otherwise better change `mb_byte2len` to accept an integer.